hazelcast/src/test/java/com/hazelcast/jet/core/JobUploadFailureTest.java

```diff
@@ -0,0 +1,247 @@
+                .isInstanceOf(JetException.class)
+                .hasCauseInstanceOf(FileSystemException.class);
+    }
+
+    @Test
+    public void test_client_jarUpload_whenResourceUploadIsNotEnabled() {
+        createHazelcastInstance();
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath());
+
+        assertThrows(JetException.class, () ->
+                jetService.submitJobFromJar(submitJobParameters)
+        );
+
+        assertEqualsEventually(() -> jetService.getJobs().size(), 0);
+    }
+
+    @Test
+    public void test_member_jarUpload_whenResourceUploadIsNotEnabled() {
+        HazelcastInstance hazelcastInstance = createHazelcastInstance();
+        JetService jetService = hazelcastInstance.getJet();
+
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath());
+
+        assertThrows(JetException.class, () ->
+                jetService.submitJobFromJar(submitJobParameters)
+        );
+
+        assertEqualsEventually(() -> jetService.getJobs().size(), 0);
+    }
+
+    @Test
+    public void test_jarUploadByClient_withWrongMainClassname() {
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        createHazelcastInstance(config);
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath())
+                .setMainClass("org.example.Main1");
+
+        assertThrows(JetException.class, () -> jetService.submitJobFromJar(submitJobParameters));
+
+    }
+
+
+    @Test
+    public void test_jarUploadByMember_withWrongMainClassname() {
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
+        JetService jetService = hazelcastInstance.getJet();
+
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath())
+                .setMainClass("org.example.Main1");
+
+        assertThrows(ClassNotFoundException.class, () -> jetService.submitJobFromJar(submitJobParameters));
+    }
+
+    @Test
+    public void test_jarUpload_WithIncorrectChecksum() throws IOException, NoSuchAlgorithmException {
+        Config config = smallInstanceConfig();
+        JetConfig jetConfig = config.getJetConfig();
+        jetConfig.setResourceUploadEnabled(true);
+
+        createHazelcastInstance(config);
+        HazelcastInstance client = createHazelcastClient();
```

Comment:
I think this block can be extracted to `@Before` method, WDYT?

Reply:
@Before is not appropriate because there are other tests that don't need it. But  I moved this block to a common method. I have also split this test  as JobUploadClientFailureTest and JobUploadMemberFailureTest as it was too big